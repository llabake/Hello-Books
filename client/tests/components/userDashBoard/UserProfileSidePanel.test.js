import React from 'react';

import {
  UserProfileSidePanel,
  mapStateToProps,
} from '../../../components/userDashBoard/UserProfileSidePanel.jsx';

import adminData from '../../mocks/userData';
import userData from '../../mocks/userData';


function setup() {
  const props = {
    dispatch: jest.fn(),
    user: adminData.admin,
    profile: userData.userProfileResponse.profile
  };

  const wrapper = shallow(<UserProfileSidePanel {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('UserProfilePage page', () => {
  test('should render and not crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  })

  test('should return an object with props required by component from store', () => {
    const store = {
      profile: {}
    };
    expect(mapStateToProps({ userReducer: {} })).toHaveProperty('profile');
  });


}) 