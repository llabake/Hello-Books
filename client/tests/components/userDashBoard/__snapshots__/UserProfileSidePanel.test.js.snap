// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfilePage page should render and not crash 1`] = `
<div
  className="col s12 m4 profile-bio "
>
  <div
    className="card large hide-on-small-only "
  >
    <div
      className="row"
    >
      <div
        className="card-image"
      >
        <img
          className="responsive-img"
          src="https://res.cloudinary.com/sardaunan/image/upload/v1520868439/rrtw8mjav6q9ntjnysvkg.jpg"
          style={
            Object {
              "marginTop": 34,
            }
          }
        />
      </div>
    </div>
    <div
      className="card-action"
    >
      <Link
        replace={false}
        to="/editprofile"
      >
        Profile Setting
      </Link>
    </div>
  </div>
  <div
    className="hide-on-med-and-up"
  >
    <ul
      className="collapsible "
      data-collapsible="accordion"
    >
      <li>
        <div
          className="collapsible-header "
        >
          <i
            className="material-icons"
          >
            account_circle
          </i>
          Profile
        </div>
        <div
          className="collapsible-body"
        >
          <span>
            <div
              className="card large"
            >
              <div
                className="row"
              >
                <div
                  className="card-image "
                >
                  <img
                    src="https://res.cloudinary.com/sardaunan/image/upload/v1520868439/rrtw8mjav6q9ntjnysvkg.jpg"
                    style={
                      Object {
                        "borderRadius": "50%",
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="card-content"
              >
                <span
                  className="card-title"
                >
                  User Bio
                </span>
                <p>
                   
                  <b>
                    Name: 
                  </b>
                   
                  keinzy
                   
                </p>
              </div>
              <div
                className="card-action"
              >
                <Link
                  replace={false}
                  to="/editprofile"
                >
                  Profile Setting
                </Link>
              </div>
            </div>
            .
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
